### ISSUE TITLE
Accessing Azure response code in Odata Client

### ISSUE THREAD

**Issue by RajatNair** [davidmoten/odata-client#23]
Hi David,

The [checkResponseCode](https://github.com/davidmoten/odata-client/blob/master/odata-client-runtime/src/main/java/com/github/davidmoten/odata/client/internal/RequestHelper.java#L84) method in `RequestHelper.java` is suppressing the actual HTTP status code from MS Graph API and that value is now part of ClientException message. To access this value, we will need to parse ClientException message. 
```Java
public static void checkResponseCode(String url, HttpResponse response,
            int expectedResponseCodeMin, int expectedResponseCodeMax) {
        if (response.getResponseCode() < expectedResponseCodeMin
                || response.getResponseCode() > expectedResponseCodeMax) {
            throw new ClientException("responseCode=" + response.getResponseCode() + " from url="
                    + url + ", expectedResponseCode in [" + expectedResponseCodeMin + ", "
                    + expectedResponseCodeMax + "], message=\n" + response.getText());
        }
    }
```

Consider this use case, a SPA client makes a call to Odata client based REST service and passes an invalid parameter. When this invalid parameter is passed to MS Graph API, it throws a HTTP 400, with a message body stating reasons for this HTTP code. I would like to pass `response.getResponseCode()` and `response.getText()` back to the client. 
>responseCode=400 from url=https://graph.microsoft.com/v1.0/organization/xyz, expectedResponseCode in [200, 200]

Any specific reason this pattern is used?

**Comment by davidmoten** [comment]
Fixed in 0.1.19 which is on Maven Central now.

**Comment by RajatNair** [comment]
Works perfectly! Thanks.
